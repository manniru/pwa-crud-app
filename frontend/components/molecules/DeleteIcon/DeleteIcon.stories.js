import { action } from '@storybook/addon-actions';
import allCombos from '../../../.storybook/allCombos';
import DeleteIcon from './DeleteIcon';

export default { title: 'Components|Molecules', component: DeleteIcon };

const data = {
  id: '1',
};

const actions = {
  onClick: action('onClick'),
};

export const deleteIcon = () => allCombos(DeleteIcon, data, actions);
