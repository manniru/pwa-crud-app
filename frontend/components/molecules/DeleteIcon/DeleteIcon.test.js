import { render, cleanup } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import { ThemeProvider } from 'styled-components';
import DeleteIcon from './DeleteIcon';
import {
  MOCK_CURRENT_USER_PAGINATED_IDEAS,
  MOCK_DELETE_IDEA,
} from '../../../__tests__/__mocks__/graphql/idea';
import theme from '../../../public/styles/theme.style';

const setup = (updatedProps = {}, updatedQueries = []) => {
  const initialProps = { id: '1' };
  const initialQueries = [MOCK_CURRENT_USER_PAGINATED_IDEAS, MOCK_DELETE_IDEA];
  const mergedQueries = updatedQueries ? updatedQueries : initialQueries;
  const mergedProps = { ...initialProps, ...updatedProps };

  const result = render(
    <MockedProvider mocks={mergedQueries} addTypename={false}>
      <ThemeProvider theme={theme}>
        <DeleteIcon {...mergedProps} />
      </ThemeProvider>
    </MockedProvider>
  );

  const deleteIcon = () => result.queryByLabelText('delete idea');

  return { ...result, mergedProps, deleteIcon };
};

describe('DeleteIcon', () => {
  afterEach(cleanup);

  it('matches snapshot', () => {
    const utils = setup();

    expect(utils.baseElement).toMatchSnapshot();
  });

  it('renders elements', () => {
    const utils = setup();

    expect(utils.deleteIcon()).toBeInTheDocument();
  });
});
