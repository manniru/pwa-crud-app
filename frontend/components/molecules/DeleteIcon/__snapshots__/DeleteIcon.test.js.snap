// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteIcon matches snapshot 1`] = `
.c0 {
  border: 1px solid rgb(117,117,117);
  height: 20px;
  width: 20px;
  cursor: pointer;
  padding: 0;
  background-color: rgb(235,245,255);
}

.c0:hover {
  background-color: #c8dcf0;
}

.c0:active {
  padding: 1px;
}

.c0 svg {
  fill: #1890ff;
  margin: 3px;
}

<body>
  <div>
    <button
      aria-label="delete idea"
      class="c0"
    >
      <svg
        viewBox="0 0 492 492"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M300.2 246L484.1 62c5.1-5.1 7.9-11.8 7.9-19 0-7.2-2.8-14-7.9-19L468 7.9C462.9 2.8 456.2 0 449 0c-7.2 0-14 2.8-19 7.9L246 191.8 62 7.9C56.9 2.8 50.2 0 43 0c-7.2 0-14 2.8-19 7.9L7.9 24c-10.5 10.5-10.5 27.6 0 38.1L191.8 246 7.9 430C2.8 435.1 0 441.8 0 449c0 7.2 2.8 14 7.9 19L24 484.1c5.1 5.1 11.8 7.9 19 7.9 7.2 0 14-2.8 19-7.9l184-184 184 184c5.1 5.1 11.8 7.9 19 7.9 7.2 0 14-2.8 19-7.9l16.1-16.1c5.1-5.1 7.9-11.8 7.9-19 0-7.2-2.8-14-7.9-19L300.2 246z"
        />
      </svg>
    </button>
  </div>
</body>
`;
